### PR TITLE
Add searchable snapshots index template for snapshot cache indices

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds the basic infrastructure that creates an index template for the future `.snapshots-N` system indices. Those indices will contain documents about blobs stored in remote snapshot repositories.

This change will go in master first and will be backported along with future changes once the feature is stable enough.

### CODE REVIEW

**Comment by dakrone** [comment]
@tlrx should this be using the legacy templates, given that they will be deprecated for 8.0? Since this is a new thing, I think it'd make more sense to start off using composable index templates so that it doesn't have to be updated at a later time

**Comment by tlrx** [comment]
@dakrone Thanks for spotting this. I missed the information that composable index templates will replace index templates. I'm going to update the code here to avoid using legacy templates.

File: x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "index_patterns": [
+    ".snapshots-${xpack.searchable_snapshots.template.version}"
+  ],
+  "priority": 100,
+  "version": ${xpack.searchable_snapshots.template.version},
+  "data_stream": {
+  },
+  "composed_of": [
+  ],
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json:10]: Super minor, but you don't have to specify this (it's optional)

File: x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      "_doc": {
+        "_meta": {
+          "searchable-snapshot-version": "${xpack.searchable_snapshots.version}"
+        },
+        "dynamic": false,
+        "properties": {
+        }
+      }
+    }
+  }
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json:30]: Can you add:

```json
  "_meta": {
    "managed": true,
    "description": "a good description of this template"
  }
```

To this? The `managed: true` ensures that the UI has a little "managed" badge by it

File: x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "version": ${xpack.searchable_snapshots.template.version},
+  "data_stream": {
+  },
+  "composed_of": [
+  ],
+  "template": {
+    "aliases": {
+      ".snapshots": {}
+    },
+    "settings": {
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json:15]: Should `.snapshots` indices also have `index.hidden: true` set in their settings? (I can't remember right now whether dot-prefixed indices automatically are hidden)

**Review by dakrone — APPROVED** [review]
LGTM, I left three really minor things, thanks Tanguy!

File: x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json (hunk 2 above)
Comment (tlrx) [reply x-pack/plugin/core/src/main/resources/searchable-snapshots-template.json:30]: Good to know, thanks!

**Review by DaveCTurner — COMMENTED** [review]
I left one request regarding naming; otherwise this is all quite unfamiliar territory to me but looks reasonable.

**Comment by tlrx** [comment]
We decided to use a dedicated system index, implemented in #60522. I'm closing this issue.

Thanks for the feedback, and sorry for the noise :pray:

**Pull request closed without merging**